### PR TITLE
upgrade(installer): Improve APM injection sockets

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
- Moves sockets to `/var/run/datadog/*.socket`
- Override the deb agent units to not have a RuntimeConfig (this stops deletion of /var/run/datadog on unit stop)
- Remove `systemd` dependency on the injector install
- Adds symlinks for SysVInit variable override as a best effort

### CODE REVIEW

File: pkg/fleet/installer/service/apm_inject.go [hunk 1]
```diff
@@ -125,20 +125,33 @@ func (a *apmInjectorInstaller) Setup(ctx context.Context) error {
...
+	if err := os.Symlink(envFilePath, "/etc/default/datadog-agent"); err != nil && !os.IsExist(err) {
+		return fmt.Errorf("failed to symlink %s to /etc/default/datadog-agent: %w", envFilePath, err)
 	}
-	if err := addSystemDEnvOverrides(ctx, traceAgentUnit); err != nil {
-		return err
-	}
-	if err := addSystemDEnvOverrides(ctx, traceAgentExp); err != nil {
-		return err
+	systemdRunning, err := isSystemdRunning()
+	if err != nil {
```
Comment (raphaelgavache) [inline pkg/fleet/installer/service/apm_inject.go:136]: if it's sysv we don't want to err

Comment (BaptisteFoy) [reply pkg/fleet/installer/service/apm_inject.go:136]: if it's sysv we're supposed to have `systemdRunning=false` and `err=nil`

**Review by raphaelgavache — APPROVED** [review]
nice, lgtm!

**Pull request merged**